src/app/kbv/controllers/single-amount-question.test.js

```diff
@@ -0,0 +1,221 @@
+
+      expect(superLocals).toHaveBeenCalledWith(req, res, expect.any(Function));
+    });
+
+    it("should set question.label and question.hint in locals", (done) => {
+      presenters.questionToLabel.mockReturnValue("Question Label");
+      presenters.questionToHint.mockReturnValue("Question Hint");
+
+      const callback = jest.fn((err, locals) => {
+        expect(err).toBeNull();
+        expect(locals.question.label).toBe("Question Label");
+        expect(locals.question.hint).toBe("Question Hint");
+        done();
+      });
+
+      controller.locals(req, res, callback);
+    });
+
+    it("should call callback with error if super.locals returns an error", (done) => {
+      const mockError = new Error("Some error");
+
+      const callback = jest.fn((err, locals) => {
+        expect(err).toEqual(mockError);
+        expect(locals).toBeUndefined();
+        done();
+      });
+
+      const superLocals = jest.spyOn(BaseController.prototype, "locals");
+      superLocals.mockImplementation((req, res, callback) => {
+        callback(mockError);
+      });
+
+      controller.locals(req, res, callback);
+    });
+  });
+
+  describe("#saveValues", () => {
+    beforeEach(() => {
+      req.session.tokenId = "session-id";
+      req.axios.post = jest.fn();
+    });
+
+    it("should call single-amount-question endpoint", async () => {
+      req.axios.get = jest.fn().mockResolvedValue({});
+      req.axios.post = jest.fn().mockResolvedValue({});
+
+      await controller.saveValues(req, res, next);
+
+      expect(req.axios.get).toHaveBeenCalledWith(QUESTION, {
+        headers: {
+          "session-id": req.session.tokenId,
+          session_id: req.session.tokenId,
+        },
+      });
+    });
+
+    describe("on API success", () => {
+      it("should call post and get api", async () => {
+        req.axios.get = jest.fn().mockResolvedValue({});
+        req.axios.post = jest.fn().mockResolvedValue({});
+
+        await controller.saveValues(req, res, next);
+
+        expect(req.axios.post).toHaveBeenCalledTimes(1);
+        expect(req.axios.get).toHaveBeenCalledTimes(1);
+      });
+
+      it("should call answer endpoint to post submitted answer", async () => {
+        req.session.question.questionKey = "rti-payslip-national-insurance";
+        req.body.question = "3";
+        req.axios.get = jest.fn().mockResolvedValue({});
+        req.axios.post = jest.fn().mockResolvedValue({});
+
+        await controller.saveValues(req, res, next);
+
+        expect(req.axios.post).toHaveBeenCalledWith(
+          ANSWER,
+          {
+            key: "rti-payslip-national-insurance",
+            value: "3",
+          },
+          {
+            headers: {
+              "session-id": req.session.tokenId,
+              session_id: req.session.tokenId,
+            },
+          }
+        );
+        expect(req.axios.post).toHaveBeenCalledTimes(1);
+        expect(req.axios.get).toHaveBeenCalledTimes(1);
```

Comment:
If we're just testing the post answer part in isolation we probably don't want to check the get call here?